thirdparty/kyaml/runfn/runfn.go

```diff
@@ -479,49 +417,16 @@ func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser
 			uidgid,
 		)
 		cf := &c
-		cf.Exec.FunctionConfig = api
-		cf.Exec.GlobalScope = r.GlobalScope
+		cf.Exec.FunctionConfig = fnConfig
 		cf.Exec.ResultsFile = resultsFile
 		cf.Exec.DeferFailure = spec.DeferFailure
 		return cf, nil
 	}
-	if r.EnableStarlark && (spec.Starlark.Path != "" || spec.Starlark.URL != "") {
-		// the script path is relative to the function config file
-		m, err := api.GetMeta()
-		if err != nil {
-			return nil, errors.Wrap(err)
-		}
-
-		var p string
-		if spec.Starlark.Path != "" {
-			p = filepath.ToSlash(path.Clean(m.Annotations[kioutil.PathAnnotation]))
-			spec.Starlark.Path = filepath.ToSlash(path.Clean(spec.Starlark.Path))
-			if filepath.IsAbs(spec.Starlark.Path) || path.IsAbs(spec.Starlark.Path) {
-				return nil, errors.Errorf(
-					"absolute function path %s not allowed", spec.Starlark.Path)
-			}
-			if strings.HasPrefix(spec.Starlark.Path, "..") {
-				return nil, errors.Errorf(
-					"function path %s not allowed to start with ../", spec.Starlark.Path)
-			}
-			p = filepath.ToSlash(filepath.Join(r.Path, filepath.Dir(p), spec.Starlark.Path))
-		}
-		fmt.Println(p)
-
-		sf := &starlark.Filter{Name: spec.Starlark.Name, Path: p, URL: spec.Starlark.URL}
-
-		sf.FunctionConfig = api
-		sf.GlobalScope = r.GlobalScope
-		sf.ResultsFile = resultsFile
-		sf.DeferFailure = spec.DeferFailure
-		return sf, nil
-	}
 
-	if r.EnableExec && spec.Exec.Path != "" {
+	if spec.Exec.Path != "" {
 		ef := &exec.Filter{Path: spec.Exec.Path}
 
-		ef.FunctionConfig = api
-		ef.GlobalScope = r.GlobalScope
+		ef.FunctionConfig = fnConfig
 		ef.ResultsFile = resultsFile
 		ef.DeferFailure = spec.DeferFailure
 		return ef, nil
```

Comment:
Given that `--image` or `--exec-path`, one of the two will be provided, handle the case where none of them is provided at the start to make the code complete in this default fn provider ?

Reply:
Added a check at the beginning of this function.